doc/api/packages.md

```diff
@@ -486,8 +486,17 @@ For example, a package that wants to provide different ES module exports for
 }
 ```
 
-Node.js implements the following conditions:
+Node.js implements the following conditions, listed in order from most
+specific to least specific as conditions should be defined:
 
+* `"node-addons"` - similar to `"node"` and matches for any Node.js environment.
+  This condition can be used to provide an entry point which uses native C++
+  addons as opposed to an entry point which is more universal and doesn't rely
+  on native addons. This condition can be disabled via the
+  [`--no-addons` flag][].
+* `"node"` - matches for any Node.js environment. Can be a CommonJS or ES
+  module file. _This condition should always come after `"import"` or
+  `"require"`._
```

Comment:
If it should always come after `import` or `require`, why is it listed before them here?

Reply:
It's kind of a category question as specificity isn't a single vector in reality, we're kind of projecting it as one here though. I think this advice was given before nested conditionals were fully supported as well. In theory node is more specific so should come early by the general rule, but if using alongside require and import should then use nesting or precedence.

How about:

```suggestion
  module file. _should only be used when absolutely necessary, and in most cases it isn't. If also using the `"import"` or `"require"` dual mode conditions, they should be nested under this condition._
```

Reply:
Better, but I think you need to explain "when absolutely necessary" might be necessary.

Reply:
I've updated the description to simply not call out this interaction. Previously when we had fewer conditions, it was the first example of a nested interaction, while now I think the principle of nesting and interactions of "require" and "import" likely applies to all conditions equally.